packages/flutter/lib/src/material/stepper.dart

```diff
@@ -0,0 +1,577 @@
+  /// keep track of it in a parent state.
+  ///
+  /// If null, the buton will be disabled.
+  final VoidCallback onStepCancel;
+
+  @override
+  _StepperState createState() => new _StepperState();
+}
+
+class _StepperState extends State<Stepper> {
+  List<GlobalKey> _keys;
+
+  @override
+  void initState() {
+    super.initState();
+    _keys = new List<GlobalKey>.generate(
+      config.steps.length,
+      (int i) => new GlobalKey()
+    );
+  }
+
+  @override
+  void didUpdateConfig(Stepper oldConfig) {
+    super.didUpdateConfig(oldConfig);
+    if (config.steps.length != oldConfig.steps.length)
+      _keys = new List<GlobalKey>.generate(
+        config.steps.length,
+        (int i) => new GlobalKey()
+      );
+  }
+
+  bool _first(int index) {
+    return index == 0;
+  }
+
+  bool _last(int index) {
+    return config.steps.length - 1 == index;
+  }
+
+  bool _current(int index) {
+    return config.currentStep == index;
+  }
+
+  Widget _line(bool visible) {
+    return new SizedBox(
+      width: visible ? 1.0 : 0.0,
+      height: 16.0,
+      child: new Container(
+        decoration: new BoxDecoration(
+          backgroundColor: Colors.grey[400]
+        )
+      )
+    );
+  }
+
+  Widget _buildCircle(BuildContext context, int index) {
+    Widget container;
+
+    if (config.steps[index].state != StepState.error) {
+      container = new AnimatedContainer(
+        curve: Curves.fastOutSlowIn,
+        duration: kThemeAnimationDuration,
+        decoration: new BoxDecoration(
+          backgroundColor: config.steps[index].state == StepState.active ?
+              Theme.of(context).primaryColor : Colors.black38,
+          shape: BoxShape.circle
+        ),
+        child: new Center(
+          child: config.steps[index].circleChild ??
+              new Text('${index + 1}', style: _kStepStyle)
+        )
+      );
+    } else {
+      container = new Center(
+        child: new SizedBox(
+          width: 24.0,
+          height: 20.78, // Height of 24dp-long-sided equilateral triangle.
+          child: new CustomPaint(
+            painter: new _TrianglePainter(
+              color: Colors.red[500]
+            ),
+            child: new Align(
+              alignment: const FractionalOffset(0.5, 0.9), // 0.9 looks better than the geometrical 0.66.
+              child: config.steps[index].circleChild ??
+                  new Text('!', style: _kStepStyle)
+            )
+          )
+        )
+      );
+    }
+
+    return new Container(
+      margin: const EdgeInsets.symmetric(vertical: 8.0),
+      child: new SizedBox(
+        width: 24.0,
+        height: 24.0,
+        child: container
+      )
+    );
+  }
+
+  Widget _buildVerticalControls() {
+    return new Container(
+      margin: const EdgeInsets.only(top: 16.0),
+      child: new ConstrainedBox(
+        constraints: const BoxConstraints.tightFor(height: 48.0),
+        child: new Row(
```

Comment:
Mostly because all the styling for the buttons is already implicitly right. Creating a `ButtonTheme` here would require styling buttons explicitly and with the `8dp` margin still required.